iptables/table.go

```diff
@@ -382,13 +388,20 @@ func NewTable(
 		countNumLinesExecuted: countNumLinesExecuted.WithLabelValues(fmt.Sprintf("%d", ipVersion), name),
 	}
 
-	if ipVersion == 4 {
-		table.iptablesRestoreCmd = "iptables-restore"
-		table.iptablesSaveCmd = "iptables-save"
-	} else {
-		table.iptablesRestoreCmd = "ip6tables-restore"
-		table.iptablesSaveCmd = "ip6tables-save"
+	verInfix := ""
+	if ipVersion == 6 {
+		verInfix = "6"
+	}
+	nftInfix := ""
+	if options.NftablesMode {
+		log.Info("Using the nftables backend for iptables.")
+		nftInfix = "nft-"
+		table.nftablesMode = true
 	}
+
+	table.iptablesRestoreCmd = "ip" + verInfix + "tables-" + nftInfix + "restore"
+	table.iptablesSaveCmd = "ip" + verInfix + "tables-" + nftInfix + "save"
```

Comment:
No, iptables-restore is the currently-selected alternative.  I should beef up this logic a bit so that legacy really means "-iptables-legacy-restore" if it's available (falling back to `iptables-restore` if not`).